### ISSUE TITLE
An error in a mariadb `docker-entrypoint-initdb.d` script does not throw an error or fail the init

### ISSUE THREAD

**Issue by rocketeerbkw** [uselagoon/lagoon-images#412]
A buggy [migration](https://github.com/uselagoon/lagoon/blob/main/services/api-db/docker-entrypoint-initdb.d/01-migrations.sql) for the lagoon `api-db` caused initialization of the database to end up in a bad state and fail the tests. It took a lot longer to figure out the error was a migration because the initialization didn't error out, it just caused a reboot (and at that point the data dir exists so it doesn't run migrations a second time).

I expect that a if a SQL statement fails, the [mariadb init should also fail](https://github.com/uselagoon/lagoon-images/blob/main/images/mariadb/entrypoints/9999-mariadb-init.bash#L132-L144).

**Comment by smlx** [comment]
Would this also be fixed by using proper migrations and versioning of the DB schema as per https://github.com/uselagoon/lagoon/issues/2995 ?

**Comment by rocketeerbkw** [comment]
Yes, this won't be a problem for the `api-db` in the long term, but that still leaves the underlying issue there to ruin the day for other users of this feature.
